code/modules/jobs/job_types/gimmick.dm

```diff
@@ -152,3 +160,141 @@
+	still_recharging_msg
+
+/obj/effect/proc_holder/spell/targeted/mail_track/Initialize(mapload)
+	. = ..()
+	var/static/casting_clothes_special
+	if(!length(casting_clothes_special))
+		casting_clothes_special = typecacheof(list(
+			/obj/item/clothing/head/mailman,
+			/obj/item/clothing/head/helmet/space/plasmaman/mailman,
+			/obj/item/clothing/under/plasmaman/mailman,
+			/obj/item/clothing/under/misc/mailman,
+			/obj/item/clothing/shoes/laceup))
+
+	casting_clothes = casting_clothes_special
+
+/obj/effect/proc_holder/spell/targeted/mail_track/cast(list/targets, mob/living/user = usr)
+	var/obj/item/mail/mail = user.get_active_held_item()
+	if(!istype(mail, /obj/item/mail))
+		to_chat(user,"<span class='notice'>You need to hold a mail on your hand.</span>")
+		charge_counter = charge_max // that activation must be mistake. fully recharges.
+		return
+	unglue_mail(mail) //failsafe when this mail becomes undelivable
+	var/datum/mind/recipient = mail.recipient_ref?.resolve()
+	if(!recipient)
+		to_chat(user,"<span class='notice'>It looks this mail goes nowhere.</span>")
+		return
+	var/mob/living/recipient_body = recipient?.current
+	if(!recipient_body)
+		to_chat(user,"<span class='notice'>You think this mail can't be delivered.</span>")
+		return
+	if(!recipient_body.stat == DEAD)
+		to_chat(user,"<span class='notice'>You sense of a danger... Better put this alone.</span>")
+		return
+	if(recipient_body == user)
+		to_chat(user,"<span class='notice'>A funny idea...</span>")
+		return
+	if(user.get_virtual_z_level() != recipient_body.get_virtual_z_level())
+		to_chat(user,"<span class='notice'>Its recipient seems to be far away.</span>")
+		return
+
+	glue_mail(mail)
+	switch(get_dist(user, recipient_body))
+		if(0 to 7)
+			unglue_mail(mail)
+			to_chat(user,"<span class='notice'>You feel your duty ends here.</span>")
+		if(8 to 15)
+			to_chat(user,"<span class='notice'>They must be somewhere here...</span>")
+		else // if you're too far away from them, your supernatural sense of direction isn't reliable
+			var/location = "nowhere"
+			if(prob(3)) // 3% chance to find a specific area.
+				location = lowertext(get_area_name(recipient_body, TRUE))
+			else
+				var/static/fake_locations = list(
+					"turn around", "go back", "behind the wall", "inside of a locker",
+					"arrival pod", "brig", "bridge", "medbay central", "cargo bay",
+					"research division", "somewhere hallway", "centcom", "lavaland mining office",
+					"spirit realm", "bearspace", "nullspace", "hyperspace", "voidspace",
+					"laghter demon's belly", "lizardpeople reptilian conspiracy association"
+				)
+				location = prob(90) ? "[dir2text(get_dir(usr, prob(90) ? pick(GLOB.player_list - user) : recipient_body))]" : pick(fake_locations)
+				// 81% (90% * 90%): tracks a proper person
+				// 9% (90% * 10%): tracks a wrong person
+				// 10%: shows improper location
+			to_chat(user,"<span class='notice'>Hmm... [lowertext(location)]?</span>")
```

Comment:
I feel like this could use more 'random' text. It could be "Maybe...", "Looks like...", "I think it's..."

Reply:
updated.